biopet-core/src/test/scala/nl/lumc/sasc/biopet/core/report/MultisampleReportBuilderTest.scala

```diff
@@ -41,10 +45,25 @@ class MultisampleReportBuilderTest extends TestNGSuite with Matchers {
 
       def libraryPage(sampleId: Int, libraryId: Int, args: Map[String, Any]) = ReportPage(Nil, Nil, Map())
     }
+    val dbFile = File.createTempFile("summary.", ".db")
+    dbFile.deleteOnExit()
+    val db = SummaryDb.openSqliteSummary(dbFile)
+    db.createTables()
+
+    val sample = Some("sampleName")
+    val lib = Some("libName")
+
+    sample.foreach { sampleName =>
+      val sampleId = Await.result(db.createSample(sampleName, 0), Duration.Inf)
```

Comment:
Not completely true. By doing Await the current thread is mark open for other tasks. So if your program is running a lot of tasks/futures/.par this is being done meanwhile.

For the testing this is indeed not going in parallel but as you already mention for testing this is less important.

Where is being used the most in the report. Here each page is already there own thread/task. So doing Await is only positive so the cpu is free to do other stuff ;).